packages/hathor-rpc-handler/src/rpcMethods/sendNanoContractTx.ts

```diff
@@ -55,6 +61,20 @@ export async function sendNanoContractTx(
     throw new Error('Neither blueprint id or ncId available');
   }
 
+  // Validate that the blueprintId exists
+  if (blueprint_id) {
+    try {
+      // Just fetch it to check if it's a valid blueprint
+      await ncApi.getBlueprintInformation(blueprint_id);
+    } catch (error) {
+      if (error instanceof NanoRequest404Error) {
+        throw new BluePrintNotFoundError();
+      }
+
+      throw new ParamsValidationError();
+    }
+  }
+
```

Comment:
We should also check that the `nc_id` (if not executing a `initialize` method) is a valid nano_contract_id

Reply:
todo: I believe it is lacking a validation on the methods existence after confirming that the blueprint exists.